tests/testthat/test-fixed_regex_linter.R

```diff
@@ -150,38 +150,38 @@ test_that("1- or 2-width octal escape sequences are handled", {
   linter <- fixed_regex_linter()
   msg <- rex::rex("For static regular expression patterns, set `fixed = TRUE`.")
 
-  expect_lint('strsplit(x, "\\1")', msg, linter)
+  expect_lint('strsplit(x, "\\\\1")', msg, linter)
```

Comment:
sorry I missed this -- I don't think this is right. What we want is

```
strsplit(x, "\1")
```

i.e., the string literal for the octal escape 1:

```
identical("\1", rawToChar(as.raw(1)))
# [1] TRUE
```

This adjusted test is actually a regex (well, not really for `strsplit()` _per se_), being the group reference:

```
grepl("(abc)def\\1", "abcdefabc")
```

IINM the `()` will always make sure it gets recognized as regex.

The version here is actually an error

```
strsplit(letters, "\\1")
# Error in strsplit("a\\1b", "\\1") : 
#   invalid regular expression '\1', reason 'Invalid back reference'
```

Reply:
Can we then use printable examples?
e.g. `[\xA7]` fails because `\xA7` dosen't match `.`

Also, the escaped versions should be added for some, e.g. `\u` escapes are also valid in regex.

Otherwise I don't know how to match "literally any unicode codepoint that exists".

Reply:
maybe it got lost in the comment stack, but here's what to match:

https://github.com/r-lib/lintr/pull/1032#issuecomment-1094218015

```
\[.\]
|
\[[\].\]
|
\[[\]x[0-9a-fA-F]{1,2}\]
|
\[[\][0-7]{1,3}\]
|
\[[\]u[0-9a-fA-F]{1,4}\]
|
\[[\]u\{[0-9a-fA-F]{1,4}\}\]
|
\[[\]U[0-9a-fA-F]{1,8}\]
|
\[[\]U\{[0-9a-fA-F]{1,8}\}\]
```

Reply:
That's what I do match, but _after_ R escaping, i.e. as literal regex characters.

Reply:
Exactly. `as.character(parse())` to be precise. I'm at least aware of `\uXXXX` being a valid escape for regex as well, so some of the `\\\\` variants should lint too. It seems there are currently too many allowed escapes (since `\1` is a backreference and not an octal escape in regex).

Reply:
> I'm at least aware of \uXXXX being a valid escape for regex as well

Could you give an example? In `gsub("\u0001", "", x)`, the regex engine only sees the byte value -- it has no idea `\u` was ever written since that is handled by the R parser.

Reply:
Compare

```
gsub(".*([abc]).*", "\1", "0123abcdef")
# "\001"
gsub(".*([abc]).*", "\\1", "0123abcdef")
# "c"
```

Reply:
Hm the docs seem not to be correct regarding some escapes, but e.g. hex escapes do work:

```
> gsub("\\x80", "", "\x80")
[1] ""
```